text_generation/causal_lm/cpp/continuous_batching/library/src/continuous_batching_pipeline.cpp

```diff
@@ -147,6 +147,14 @@ class ContinuousBatchingPipeline::Impl {
             timer.end();
         }
 
+        // if no tokens were scheduled, we are out of memory
+        if (scheduler_output.m_total_num_scheduled_tokens == 0) {
+            for (size_t sequence_group_id = 0; sequence_group_id < m_requests.size(); ++sequence_group_id) {
+                m_requests[sequence_group_id]->set_out_of_memory();
```

Comment:
maybe we need to drop sequence from generation at all if it requires more memory than we have?
e.g. GenerationResult can have additional status for requests:
- FINISHED - generation is successfully performed
- IGNORED / OUT_OF_MEMORY - generation requires more memory than we have in KV cache space
- ABORTED - request is aborted by user. Currently, we don't have such API, but in future I suppose we need functionality to drop generation by request_id for example when an user of OVMS was disconnected and we can drop its request. 

E.g. vLLM has the following code to ignore sequence (out of memory in our case):
```python
            # If the sequence group cannot be allocated, stop.
            can_allocate = self.block_manager.can_allocate(seq_group)
            if can_allocate == AllocStatus.LATER:
                break
            elif can_allocate == AllocStatus.NEVER:
                logger.warning(
                    "Input prompt (%d tokens) is too long"
                    " and exceeds the capacity of block_manager",
                    num_new_tokens)
                for seq in waiting_seqs:
                    seq.status = SequenceStatus.FINISHED_IGNORED
                ignored_seq_groups.append(seq_group)
                waiting_queue.popleft()
                continue
```
We could do similar handling

Reply:
Added `GenerationResultStatus`, please take a look.